Framework/Algorithms/src/CorelliPowderCalibrationApply.cpp

```diff
@@ -0,0 +1,193 @@
+    issues["Workspace"] = "CORELLI only algorithm, aborting";
+  }
+
+  // 2_check: headers of calibration table
+  calTable = getProperty("CalibrationTable");
+  auto refCalTableHeader = CorelliCalibration::calibrationTableColumnNames;
+  std::vector<std::string> colnames = calTable->getColumnNames();
+  if (colnames.size() != refCalTableHeader.size()) {
+    issues["CalibrationTable"] =
+        "Headers of input calibration table does not match required";
+  }
+  for (size_t i = 0; i < colnames.size(); i++) {
+    if (colnames[i] != refCalTableHeader[i]) {
+      issues["CalibrationTable"] = "Header mismatch at " + std::to_string(i);
+      break;
+    }
+  }
+
+  return issues;
+}
+
+/**
+ * @brief Executes the algorithm.
+ *
+ */
+void CorelliPowderCalibrationApply::exec() {
+  g_log.notice() << "Start applying CORELLI calibration\n";
+
+  // Parse input arguments
+  ws = getProperty("Workspace");
+  auto wsName = ws->getName();
+
+  calTable = getProperty("CalibrationTable");
+  auto componentNames = calTable->getColumn(0);
+  auto x_poss = calTable->getColumn(1);
+  auto y_poss = calTable->getColumn(2);
+  auto z_poss = calTable->getColumn(3);
+  auto rotxs = calTable->getColumn(4);
+  auto rotys = calTable->getColumn(5);
+  auto rotzs = calTable->getColumn(6);
+  auto rotangs = calTable->getColumn(7); // unit: degrees
+
+  /**
+  Translate each component in the instrument
+  [moderator, sample-position, bank1,.. bank92]
+  NOTE:
+  1. moderator: this is often called "source" in other settings, but CORELLI
+                instrument uses the term moderator for it.
+  2. sample-position: this is often referred to as "sample" in other
+                      instrument definition, but the CORELLI instrument uses
+                      a more verbose version here.
+  3. the tranlation vector calcuated in upstream is the absolute coordinates
+     of each component (in lab frame), therefore we need to toggle the option
+     RelativePosition off here.
+  */
+  g_log.notice() << "Translating each component using given Calibration table";
+  auto moveAlg = createChildAlgorithm("MoveInstrumentComponent");
+  moveAlg->initialize();
+  moveAlg->setProperty("Workspace", wsName);
+  auto componentName = calTable->getColumn(0);
+  for (size_t row_num = 0; row_num < calTable->rowCount(); row_num++) {
+    moveAlg->setProperty("ComponentName",
+                         componentNames->cell<std::string>(row_num));
+    moveAlg->setProperty("X", x_poss->cell<double>(row_num));
+    moveAlg->setProperty("Y", y_poss->cell<double>(row_num));
+    moveAlg->setProperty("Z", z_poss->cell<double>(row_num));
+    // [IMPORTANT]
+    // The position data from calibration table are ABSOLUTE values (lab frame)
+    moveAlg->setProperty("RelativePosition", false);
+    moveAlg->execute();
+  }
+
+  /**
+  Rotate each component in the instrument
+  IMPORTANT NOTE:
+  0. The upstream calcualtion is a minimization routine that searches the
```

Comment:
small typo in "calcualtion"

Reply:
opps :sweat_smile: 
corrected in the next batch of commit